src/Illuminate/Support/HigherOrderCollectionProxy.php

```diff
@@ -54,7 +54,15 @@ public function __get($key)
     public function __call($method, $parameters)
     {
         return $this->collection->{$this->method}(function ($value) use ($method, $parameters) {
-            return $value->{$method}(...$parameters);
+            if (is_object($value)) {
+                return $value->{$method}(...$parameters);
+            }
+
+            if ($method == 'empty') {
```

Comment:
Exactly, you cannot call empty with as a dynamic / variable function (PHP limitation), same with `echo`, `unset` and others, but from that list empty imo is the only important one to support.